### DISCUSSION TITLE
Multipoint optimisation of an airfoil

### DISCUSSION THREAD

**Discussion by Arunprasath1994** [mdolab/dafoam#337] [General]
I am Arun from Plana and I am writing to you regarding DAfoam. I am trying to run a multipoint optimisation using Dafoam for an airfoil. Upon using the tutorial case (runscript_v2 as the base), I get errors.

1. Cannot project the nodes at XXX to normal YYY. I have changed the airfoil coordinates at the trailing edge too and tried to play around with the TElist. This problem doesn't happen anymore but I really don't understand the purpose of the LElist and TElist
2. Once this is solved, for some reason when the decomposePar is used, the initial 0 folder files are not decomposed into the individual processors. I get the error 

---[0] --> FOAM FATAL ERROR:
[0] cannot find file "/home/dafoamuser/mount/processor0/0/p"
or
Unable to run decomposePar, the domain has been already decomposed?
or
--> FOAM FATAL IO ERROR:
Cannot find patchField entry for auto5

I am attaching the airfoil coordinates file for your reference. 
Is there anyone who has a modified runscript for multipoint optimisation of an airfoil?

[NACA0012PS1.txt](https://github.com/mdolab/dafoam/files/9421507/NACA0012PS1.txt)
[NACA0012SS1.txt](https://github.com/mdolab/dafoam/files/9421508/NACA0012SS1.txt)

**Comment by friedenhe** [comment]
Try to run the multipoint tutorial without any change. Make sure it works. Then add your changes one by one and see which one breaks it. Also, I suggest you use version 3. The multipoint is much more flexible in v3.

**Reply by Arunprasath1994** [reply]
Hello, I am using the version 3. The tutorial case runs fine with NACA0012. I have just changed airfoil coordinates to S7012 and I have the error that it Cannot project the nodes at XXX to normal YYY. In the documentation, I see that it is due to improper LeList and Telist specification. Could you please tell me how to go about that?

**Reply by friedenhe** [reply]
Check 2021 workshop on how to set leList and teList.
